python/LoggingWrapper.cpp

```diff
@@ -1,57 +1,159 @@
 #include <string>
 
 #include "../util/Logger.h"
-#include <boost/log/trivial.hpp>
 
+#include <boost/log/expressions.hpp>
+#include <boost/log/support/date_time.hpp>
 #include <boost/python.hpp>
 
+namespace expr = boost::log::expressions;
+
 namespace {
     // Expose interface for redirecting standard output and error to FreeOrion
     // logging.  Can be imported before loading the main FreeOrion AI interface
     // library.
-    static const int MAX_SINGLE_CHUNK_TEXT_SIZE = 4096;
-
-    // stdout logger
-    static std::string log_buffer("");
-    void LogText(const char* text) {
-        // Python sends text as several null-terminated array of char which need to be
-        // concatenated before they are output to the logger.  There's probably a better
-        // way to do this, but I don't know what it is, and this seems reasonably safe...
-        if (!text) return;
-        for (int i = 0; i < MAX_SINGLE_CHUNK_TEXT_SIZE; ++i) {
-            if (text[i] == '\0') break;
-            if (text[i] == '\n' || i == MAX_SINGLE_CHUNK_TEXT_SIZE - 1) {
-                DebugLogger() << log_buffer;
-                log_buffer = "";
-            } else {
-                log_buffer += text[i];
-            }
+    static const std::size_t MAX_SINGLE_CHUNK_TEXT_SIZE = 4096;
+
+    // Python sends text as several null-terminated array of char which need to be
```

Comment:
Is this comment relevant now? The parameter type is no longer `char *` Does the `std::string input` still have null characters in it? If not, then the search within it seems unnecessary...